databricks/koalas/indexes.py

```diff
@@ -123,6 +123,31 @@ def size(self) -> int:
         """
         return len(self._kdf)  # type: ignore
 
+    @property
+    def shape(self) -> tuple:
+        """
+        Return a tuple of the shape of the underlying data.
+
+        Examples
+        --------
+        >>> idx = ks.Index(['a', 'b', 'c'])
+        >>> idx
+        Index(['a', 'b', 'c'], dtype='object')
+        >>> idx.shape
+        (3,)
+
+        >>> midx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
+        >>> midx  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y'),
+                    ('c', 'z')],
+                   )
+
+        >>> midx.shape
+        (3,)
+        """
+        return len(self._kdf),
```

Comment:
@RainFung, can you open another PR to implement `def __len__` here?

And then, we can move this implementation of `shape` from here and `Series to `IndexOpsMixin`.